specification/_global/health_report/types.ts

```diff
@@ -166,3 +167,19 @@ export class SlmIndicatorUnhealthyPolicies {
   count: long
   invocations_since_last_success?: Dictionary<string, long>
 }
+
+/** SHARDS_CAPACITY */
+
+export class ShardsCapacityIndicator extends BaseIndicator {
+  details?: ShardsCapacityIndicatorDetails
+}
+
+export class ShardsCapacityIndicatorDetails {
+  data: ShardsCapacityIndicatorTierDetail
+  frozen: ShardsCapacityIndicatorTierDetail
+}
```

Comment:
This pattern where all fields are of the same type could also be a `Dictionary<string, SomeType>` to allow for future expansion, or also to account for values we haven't seen in the docs or tests.

In this particular case, looking at the ES code shows that these are hardcoded constants that are unlikely to have new members. So this approach works.